### PR TITLE
Add firmware volume parser

### PR DESCRIPTION
**Pull request by rjoleary**
Signed-off-by: Ryan O'Leary <ryanoleary@google.com>

### CODE REVIEW

**Comment by rjoleary** [comment]
Use https://github.com/xoac/endian_codec instead of unsafe.

**Comment by rjoleary** [comment]
> Use https://github.com/xoac/endian_codec instead of unsafe.

Or https://crates.io/crates/packed_struct

**Comment by rjoleary** [comment]
> > Use https://github.com/xoac/endian_codec instead of unsafe.
> 
> Or https://crates.io/crates/packed_struct

https://users.rust-lang.org/t/how-to-derive-custom-traits-for-struct-definitions-generated-by-bindgen/39710 has some ideas on deriving types on bindgenerated code.

File: src/vendorcode/fsp/coffeelake/build.rs [hunk 1]
```diff
@@ -35,103 +35,103 @@ fn main() {
         // Only generate types and constants.
         .with_codegen_config(bindgen::CodegenConfig::TYPES | bindgen::CodegenConfig::VARS)
         // Whitelist of types and constants to import.
```
Comment (rminnich) [inline src/vendorcode/fsp/coffeelake/build.rs:37]: Whitelist -> allowlist in comment.

File: src/vendorcode/fsp/qemu/build.rs [hunk 2]
```diff
@@ -43,37 +43,39 @@ fn main() -> std::io::Result<()> {
...
         // Tell cargo to invalidate the built crate whenever any of the included header files
-        // changed.
+        // change.
         .parse_callbacks(Box::new(bindgen::CargoCallbacks))
         // Use core:: instead of std::
         .use_core()
+        .ctypes_prefix("cty")
         // Only generate types and constants.
         .with_codegen_config(bindgen::CodegenConfig::TYPES | bindgen::CodegenConfig::VARS)
         // Whitelist of types and constants to import.
```
Comment (rminnich) [inline src/vendorcode/fsp/qemu/build.rs:55]: same but ... we have to repeat this everywhere? Is there some common crate we could have with this (later)?

**Review by rminnich — COMMENTED** [review]
see one nitpick comment. I like this a lot, it's a good model for me to learn from.

File: src/vendorcode/fsp/coffeelake/build.rs (hunk 1 above)
Comment (rjoleary) [reply src/vendorcode/fsp/coffeelake/build.rs]: Done.

File: src/vendorcode/fsp/qemu/build.rs (hunk 2 above)
Comment (rjoleary) [reply src/vendorcode/fsp/qemu/build.rs]: Yes, this can be put in a common crate (i think) (later).

**Review by rminnich — APPROVED** [review]

**Pull request merged**